text/0000-dockerfiles.md

```diff
@@ -0,0 +1,153 @@
+# Meta
+[meta]: #meta
+- Name: Support Dockerfiles
+- Start Date: 2021-06-30
+- Author(s): sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: [RFC0069](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md), [RFC#167](https://github.com/buildpacks/rfcs/pull/167)
+- Depends on: [RFC#172](https://github.com/buildpacks/rfcs/pull/172)
+
+# Summary
+[summary]: #summary
+
+This RFC introduces functionality for customizing base images, as an alternative to stackpacks.
+
+# Motivation
+[motivation]: #motivation
+
+Relying on Dockerfiles for base image generation and manipulation allows us to apply buildpacks only to the problem that they solve best: managing application runtimes and dependencies.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes that we replace stackpacks with multi-purpose build-time and runtime Dockerfiles.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Note: kaniko, buildah, BuildKit, or the original Docker daemon may be used to apply Dockerfiles at the platform's discretion. 
+
+### Builder-specified Dockerfiles
+
+A builder may specify any number of executable "hooks" in `/cnb/hooks.d/`.
+Hook files in this directory are executed in the context of app directory and must output Dockerfiles that are applied to the build-time and runtime base images before an image is built.
+
+If a hook exits with a non-zero status value, the build fails. If a hook exits with a zero status value and no output, the hook is ignored. Directories and non-executable files are ignored.
+
+Each executable must be in the format `/cnb/hooks.d/<name>.(build.|run.|)<format>`, where build, run, or empty specify the build-time base image, runtime base image, or both bash images, respectively. The only valid format is `Dockerfile`, although support for, e.g. LLB JSON, could be added in the future. 
```

Comment:
As I envision this hook directory listing, not considering some of the good thoughts @cmoulliard has brought up, I would prefer to group the files by phase first, then order, and lastly format. 

It seems easier to reason about the order of operations:

#### Current

```
./hooks.d/1_step1.Dockerfile
./hooks.d/1_step1.build.Dockerfile
./hooks.d/1_step1.run.Dockerfile
./hooks.d/2_step2.Dockerfile
./hooks.d/2_step2.build.Dockerfile
./hooks.d/3_step3.build.Dockerfile
```

#### Suggested

```
./hooks.d/1_step1.Dockerfile
./hooks.d/2_step2.Dockerfile
./hooks.d/build.1_step1.Dockerfile
./hooks.d/build.2_step2.Dockerfile
./hooks.d/build.3_step3.Dockerfile
./hooks.d/run.1_step1.Dockerfile
```

Reply:
With the order proposed in the RFC, the list order matches the order of application.
With the order you recommend, the "both" hooks may appear out-of-order. This seems more confusing to me.

Reply:
Now that hooks -> extensions are part of an order, and have a `/cnb/extensions` directory, this is moot. I'm going to mark this as resolved.